cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -80,6 +81,7 @@ struct parse_options {
   char thousands;
   char comment;
   bool keepquotes;
+  bool detect_whitespace_around_quotes;
```

Comment:
*question*: It seems a shame that one must repeat most of the fields between `parse_options` and `parse_options_view`. Shall we open an issue to discuss whether these should be shared with something like:

```c++
struct _parse_options {
   char delimiter;
   ...;
   ...;
}

struct parse_options_view {
    struct _parse_options opts;
    cudf::detail::trie_view trie_true;
    ...;
}

struct parse_options {
   struct _parse_options opts;
   cudf::detail::optional_tree trie_true;
   ...
   
}
```
WDYT?

Reply:
@vuule @shrshi I like this idea, WDYT? Not in this PR, but could be a worthwhile follow-up.

Reply:
Reducing the redundancy sounds good to me (not for this PR).
Less intrusive way [citation needed :)] to do this might be to have `parse_options_base` with trivial members and derived `parse_options` and `parse_options_view` with additional trie members.

Reply:
I'm going to move this comment to a new issue for continuing the discussion.

Reply:
https://github.com/rapidsai/cudf/issues/15825

Reply:
Thanks @vyasr, I had not considered this earlier. Having a shared base structs will help eliminate any errors that may occur if struct fields are not properly updated.